plugins/registry/functions.go

```diff
@@ -100,9 +108,25 @@ func pushToRegistry(appName string, tag int, imageID string, imageRepo string) e
 		return errors.New("Unable to tag image")
 	}
 
-	// For the future, we should also add the ability to create the remote repository
-	// This is only really important for registries that do not support creation on push
-	// Examples include AWS and Quay.io
+	extraTags := getRegistryPushExtraTagsForApp(appName)
+	if extraTags != "" {
+		extraTagsArray := strings.Split(extraTags, ",")
+		for _, extraTag := range extraTagsArray {
+			extraTagImage := fmt.Sprintf("%s%s:%s", registryServer, imageRepo, extraTag)
+			common.LogVerboseQuiet(fmt.Sprintf("Tagging %s as %s in registry format", imageRepo, extraTag))
+			if !dockerTag(imageID, extraTagImage) {
+				return errors.New(fmt.Sprintf("Unable to tag image as %s", extraTag))
+			}
+			common.LogVerboseQuiet(fmt.Sprintf("Pushing %s", extraTagImage))
+			if !dockerPush(extraTagImage) {
+				return errors.New(fmt.Sprintf("Unable to push image with %s tag", extraTag))
+			}
+			common.LogVerboseQuiet(fmt.Sprintf("Untagging extra tag %s", extraTag))
+			if !dockerRemoveTag(extraTagImage) {
+				return errors.New(fmt.Sprintf("Unable to untag extra tag %s", extraTag))
+			}
```

Comment:
This should be a defer before line 120:

```
defer func() {
			common.LogVerboseQuiet(fmt.Sprintf("Untagging extra tag %s", extraTag))
			if !dockerRemoveTag(extraTagImage) {
				common.LogWarn(fmt.Sprintf("Unable to untag extra tag %s", extraTag))
			}
}()
```

That will guarantee it runs even if the push fails, ensuring we don't have dangling images.

Reply:
ah, right, let me correct this

Reply:
Ohh, w8 - but then it would tag and untag before `dockerPush` happens, no? Wouldn't it be better to untag it here?

```
121:			if !dockerPush(extraTagImage) {
122:				return errors.New(fmt.Sprintf("Unable to push image with %s tag", extraTag))
123:			}
```
For example:

```
			if !dockerPush(extraTagImage) {
				if !dockerRemoveTag(extraTagImage) { return errors.New(fmt.Sprintf("Unable to untag extra tag %s", extraTag)) }
				return errors.New(fmt.Sprintf("Unable to push image with %s tag", extraTag))
			}
```

Reply:
The steps should be something like:

```
docker-tag
defer docker-tag-removal
docker-push
```

If the tag is created, we immediately fire off a deferred function, and then continue as normal. If the push fails - or we encounter any errors - the tag will be removed. If everything succeeds, then the tag will _also_ be removed.

https://gobyexample.com/defer

Reply:
Ahh, see. Sorry, I am not yet fluent with Go 🙈 I am on my learning path.

Reply:
No worries, you're doing great!

I definitely recommend reading through gobyexample, its super straightforward and conveys a lot of what you need to be productive.

Reply:
I see. Updated it to run as anonymous defer function